src/runtime/providers/math.ts

```diff
@@ -24,10 +26,13 @@ abstract class TotalFunctionConstraint extends Constraint {
 
   // Total functions always have a cardinality of 1
   getProposal(tripleIndex, proposed, prefix) {
-    let proposal = this.proposalObject;
-    proposal.providing = proposed;
-    proposal.cardinality = 1;
-    return proposal;
+    if(this.returns.length) {
```

Comment:
Ah the logical providers do this for a reason that doesn't really apply here. There are two cases for the logic operators, the standard filtering case `x > y` and the value returning case using `is(x > y)`. In the former case, we never want to propose anything and instead just go through the testing phase. In the latter case, `is` adds a return variable that we'll want to fill in with the boolean value of the expression. With math ops, they should always have a bound return, so the check doesn't buy us anything.